api/src/iot/import_utils.py

```diff
@@ -30,84 +26,576 @@ def get_center_coordinates(postcode: str) -> Point:
+    pass
+
+
+class InvalidThemes(ValidationError):
+    pass
+
+
+class InvalidLatitude(ValidationError):
+    pass
+
+
+class InvalidLongitude(ValidationError):
+    pass
+
+
+class InvalidLegalGround(ValidationError):
+    pass
+
+
+class InvalidPostcode(ValidationError):
+    pass
+
+
+class InvalidHouseNumber(ValidationError):
+    pass
+
+
+class InvalidRegion(ValidationError):
+    pass
+
+
+class InvalidLocationDescription(ValidationError):
+    pass
+
+
+class InvalidDate(ValidationError):
+    pass
+
+
+def validate_sensor(sensor_data: SensorData):
+    """
+    Ensure that the sensor data contains valid data, raises an exception when
+    the data is invalid.
+    """
+    validate_type(sensor_data)
+    validate_themes(sensor_data)
+    validate_location(sensor_data)
+    validate_contains_pi_data(sensor_data)
+    validate_legal_ground(sensor_data)
+    validate_active_until(sensor_data)
+
+
+DATE_FORMAT = '%d-%m-%Y'
+
+
+def validate_active_until(sensor_data):
+    try:
+        datetime.datetime.strptime(sensor_data.active_until, DATE_FORMAT)
+    except Exception as e:
+        raise InvalidDate(sensor_data) from e
+
+
+def validate_legal_ground(sensor_data):
+    if sensor_data.contains_pi_data == 'Ja':
+        if not models.LegalGround.objects.filter(name=sensor_data.legal_ground).exists():
+            raise InvalidLegalGround(sensor_data)
+
+
+def validate_contains_pi_data(sensor_data):
+    if sensor_data.contains_pi_data not in ("Ja", "Nee"):
+        raise InvalidContainsPiData(sensor_data)
+
+
+def validate_type(sensor_data):
+    if sensor_data.type in ('', None):
+        raise InvalidSensorType(sensor_data)
+
+
+def validate_themes(sensor_data):
+    themes = (sensor_data.themes or '').split(',')
+    valid_themes = models.Theme.objects.filter(name__in=themes).count()
+    if valid_themes != len(themes):
+        raise InvalidThemes(sensor_data)
+
+
+def validate_region(sensor_data):
+    if not sensor_data.location.region:
+        raise InvalidRegion(sensor_data)
+
+
+def validate_location(sensor_data):
+    if isinstance(sensor_data.location, Region):
+        validate_region(sensor_data)
+    elif isinstance(sensor_data.location, LatLong):
+        validate_latitude_longitude(sensor_data)
+    elif isinstance(sensor_data.location, PostcodeHouseNumber):
+        validate_postcode_house_number(sensor_data)
+    elif isinstance(sensor_data.location, LocationDescription):
+        validate_location_description(sensor_data)
+
+
+def validate_location_description(sensor_data):
+    if not sensor_data.location.description:
+        raise InvalidLocationDescription(sensor_data)
+
+
+def validate_postcode_house_number(sensor_data):
+    postcode_regex = r'\d\d\d\d ?[a-zA-Z][a-zA-Z]'
+    if not sensor_data.location.postcode or \
+            not re.match(postcode_regex, sensor_data.location.postcode):
+        raise InvalidPostcode(sensor_data)
+    try:
+        int(sensor_data.location.house_number)
```

Comment:
Ik twijfel of deze validatie voldoende is. Adressen (officieel 'nummeraanduiding': https://www.amsterdam.nl/stelselpedia/bag-index/catalogus-bag/objectklasse-num/) zijn complexe objecten. 

Zo heb je naast huisnummers ook huisletters en huisnummer toevoegingen. Moet je die ook meenemen?

Een voorbeeld is mijn oude adres:
> Bataviastraat 39B. Hierin is het nummer 39, en de huisletter is B. 

Een ander voorbeeld:
> Nieuwe kerkstraat 16-1. Nummer 16, huisnummer toevoeging is 1.

Reply:
In deze voorbeelden de B en 1 zijn toevoegingen?

Het gebruik van toevoeging om de locatie nog accurater te maken is een "nice to have" dus in deze eerste versie maak ik daar geen gebruik van.